features/eager_workflow/successful_start/README.md

```diff
@@ -0,0 +1,10 @@
+# Successful start of a workflow using eager mode
+Eager Workflow Start (EWS) is a latency optimization to reduce the time to start processing the first task of a workflow. The starter program provisions a slot in a suitable worker, requests that the server starts a workflow in eager mode, and then, when it receives the first WFT in the response, it directly schedules the task to the worker, eliminating a network roundtrip and a database transaction.
+
+In each scenario, the starter program and the worker should share a client. The starter program will create a simple workflow in eager mode, and then verify that eager mode was actually used, and the first workflow task was processed correctly.
+
+# Detailed spec
+* The `EnableEagerStart` start workflow option should be `true`.
+* The server response to start workflow should include a non-nil `eager_workflow_task` field.
+* The task timeout for the workflow should be large enough to hang the program on a task retry.
```

Comment:
What is the  goal of large task timeout? If it is just to guard against task retry then why don't you test that directly? You could assert on metrics or add another atomic counter the workflow  function increments.

Reply:
Need to clarify in the spec better. The problem is that the gprc interceptors only tell us that the server returned the first task, but the worker, even if we had secured a slot beforehand, could still refuse to process it, or may have crashed...

In that exceptional case the task will be retried after the task timeout, and could eventually succeed through the non-eager path. By adding a very large task timeout we effectively disable retries, and we know that success always comes from the eager path. Atomic counter wouldn't help (it only executes once in both cases), and messing with metrics is much more complicated...

Reply:
If the worker crashed then the whole test all crashed so that would be a failure either way. I do see your point though about the worker refusing that's a good reason for the high timeout.